Update and cleanup chef provisioning test environment

ramereth opened the pull request:
The big thing here is moving the .chef folder somewhere else so that the
cookbook uploader doesn't break when we bump this cookbook.

jldugger commented:
It seems like you put some work into this; can you help me understand the context of what this is trying to fix and why we want it? I see some chef-provisioner references, but no documentation here or in docs.o.o about this tech.

ramereth commented:
@jldugger chef provisioning is currently used in a few cookbooks to test multi-node interactions with the code base. It's initiated via rake and usually documented in the README file of the cookbook. It requires a fake knife.rb in order to work since we're using chef-zero. When I initially set these up, I just put it in the .chef folder in the top of the cookbook.

This wasn't a problem until we introduced the cookbook uploader jenkins jobs. When that job runs, it tries to use the fake knife.rb in since it's in the default .chef folder location and fails to upload it to our chef server. This just moves that to a non-standard location that Rake still knows about. I also cleaned up the codebase to match what I've done more recently.

ramereth commented:
I'm just merging it as-is as it doesn't affect the cookbook itself.
